internal/service/enrollmentrequest.go

```diff
@@ -296,16 +304,19 @@ func (h *ServiceHandler) ApproveEnrollmentRequest(ctx context.Context, name stri
 		}
 	}
 	_, err = h.store.EnrollmentRequest().UpdateStatus(ctx, orgId, enrollmentReq)
-	return approvalStatusToReturn, StoreErrorToApiStatus(err, false, api.EnrollmentRequestKind, &name)
+	status := StoreErrorToApiStatus(err, false, api.EnrollmentRequestKind, &name)
```

Comment:
What about EnrollmentRequestApprovalFailed events for the failures above?

return nil, api.StatusInternalServerError(fmt.Sprintf("failed to retrieve user identity while approving enrollment request: %v", err))

return nil, api.StatusInternalServerError(fmt.Sprintf("error creating device from enrollment request: %v", err))